llvm/clang warnings: implicit conversion

These 4 warnings occur while compiling some of the source files:
```
In file included from ./nghttp2/lib/nghttp2_frame.c:33:
./nghttp2/lib/nghttp2_net.h:74:14: warning: implicit conversion changes signedness: 'int' to 'uint32_t' (aka 'unsigned int') [-Wsign-conversion]
  res = *p++ << 24;
      ~ ~~~~~^~~~~
./nghttp2/lib/nghttp2_net.h:75:15: warning: implicit conversion changes signedness: 'int' to 'unsigned int' [-Wsign-conversion]
  res += *p++ << 16;
      ~~ ~~~~~^~~~~
./nghttp2/lib/nghttp2_net.h:76:15: warning: implicit conversion changes signedness: 'int' to 'unsigned int' [-Wsign-conversion]
  res += *p++ << 8;                                      
      ~~ ~~~~~^~~~                                       
./nghttp2/lib/nghttp2_net.h:84:14: warning: implicit conversion loses integer precision: 'int' to 'uint16_t' (aka 'unsigned short') [-Wimplicit-int-conversion]
  res = *p++ << 8;                                                        
      ~ ~~~~~^~~~                                                         
4 warnings generated.                                                     
```

Compiler: llvm/clang 15.0.0 cross-compiling for Windows x64.

This looks like the same issue as in ngtcp2: https://github.com/ngtcp2/ngtcp2/issues/599

--- a/lib/nghttp2_net.h
+++ b/lib/nghttp2_net.h
@@ -71,9 +71,9 @@ STIN uint16_t htons(uint16_t hostshort) {
 STIN uint32_t ntohl(uint32_t netlong) {
   uint32_t res;
   unsigned char *p = (unsigned char *)&netlong;
-  res = *p++ << 24;
-  res += *p++ << 16;
-  res += *p++ << 8;
+  res = (uint32_t)(*p++ << 24);
+  res += (uint32_t)(*p++ << 16);
+  res += (uint32_t)(*p++ << 8);
   res += *p;
   return res;
 }
@@ -81,7 +81,7 @@ STIN uint32_t ntohl(uint32_t netlong) {
 STIN uint16_t ntohs(uint16_t netshort) {
   uint16_t res;
   unsigned char *p = (unsigned char *)&netshort;
-  res = *p++ << 8;
+  res = (uint16_t)(*p++ << 8);
   res += *p;
   return res;
 }

